### PR TITLE
add new config options for connecting to cassandra

### PR DESCRIPTION
**Pull request by woodsaj**
Some managed Cassandra cluster services, like Azure Cosmos DB, dont expose
the internal cluster topology. So we need to disable initial host lookup
and just send requests to the configured cluster address.

### CODE REVIEW

**Comment by tomwilkie** [comment]
I try and have master here track upstream; so you should probably make this PR to there?

File: pkg/chunk/cassandra/storage_client.go [hunk 1]
```diff
@@ -48,6 +62,20 @@ func (cfg *Config) session() (*gocql.Session, error) {
...
+			CaPath:                 cfg.caPath,
+			EnableHostVerification: cfg.hostVerification,
+		}
+	}
+	if cfg.auth {
+		cluster.Authenticator = gocql.PasswordAuthenticator{
+			Username: cfg.username,
+			Password: cfg.password,
+		}
+	}
```
Comment (tomwilkie) [inline pkg/chunk/cassandra/storage_client.go:86]: Would be good to extract as much of this as we can into a helper function, to reduce repetition.

**Comment by tomwilkie** [comment]
One nit, otherwise LGTM!

**Comment by woodsaj** [comment]
ok, closing this. will open a PR upstream

File: pkg/chunk/cassandra/storage_client.go (hunk 1 above)
Comment (trongtb88) [reply pkg/chunk/cassandra/storage_client.go:86]: DisableInitialHostLookup still found at java datastatx driver, right ?

**Pull request closed without merging**